genrl/deep/agents/sac/sac.py

```diff
@@ -1,453 +1,463 @@
-            next_state, reward, done, _ = self.env.step(action)
-            if self.render:
-                self.env.render()
-
-            done = [
-                False if ep_len == self.max_ep_len else done for ep_len in episode_len
-            ]
-
-            if np.any(done) or np.any(episode_len == self.max_ep_len):
-                for j, di in enumerate(done):
-                    if di:
-                        episode_reward[j] = 0
-                        episode_len[j] = 0
-
-            self.replay_buffer.extend(zip(state, action, reward, next_state, done))
-            state = next_state
-
-            if i > total_steps:
-                break
-
-            if sum(episode_len) % (5 * self.env.n_envs) == 0 and sum(episode_len) != 0:
-                print(
-                    "Episode: {}, total numsteps: {}, reward: {}".format(
-                        sum(episode_len), i, episode_reward
-                    )
-                )
-            # ep += 1
-
-        self.env.close()
-
-    def get_hyperparams(self) -> Dict[str, Any]:
-        hyperparams = {
-            "network_type": self.network_type,
-            "gamma": self.gamma,
-            "lr": self.lr,
-            "replay_size": self.replay_size,
-            "entropy_tuning": self.entropy_tuning,
-            "alpha": self.alpha,
-            "polyak": self.polyak,
-            "q1_weights": self.q1.state_dict(),
-            "q2_weights": self.q2.state_dict(),
-            "policy_weights": self.policy.state_dict(),
-        }
-
-        return hyperparams
-
-    def load_weights(self, weights) -> None:
-        """
-        Load weights for the agent from pretrained model
-        """
-        self.q1.load_state_dict(weights["q1_weights"])
-        self.q2.load_state_dict(weights["q2_weights"])
-        self.policy.load_state_dict(weights["policy_weights"])
-
-    def get_logging_params(self) -> Dict[str, Any]:
-        """
-        :returns: Logging parameters for monitoring training
-        :rtype: dict
-        """
-        logs = {
-            "policy_loss": safe_mean(self.logs["policy_loss"]),
-            "q1_loss": safe_mean(self.logs["q1_loss"]),
-            "q2_loss": safe_mean(self.logs["q2_loss"]),
-            "alpha_loss": safe_mean(self.logs["alpha_loss"]),
-        }
-
-        self.empty_logs()
-        return logs
-
-    def empty_logs(self):
-        """
-        Empties logs
-        """
-        self.logs = {}
-        self.logs["q1_loss"] = []
-        self.logs["q2_loss"] = []
-        self.logs["policy_loss"] = []
-        self.logs["alpha_loss"] = []
+from copy import deepcopy
+from typing import Any, Dict, Optional, Tuple, Union
+
+import gym
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as opt
+from torch.distributions import Normal
+
+from genrl.deep.common import (
+    BaseActorCritic,
+    ReplayBuffer,
+    get_env_properties,
+    get_model,
```

Comment:
There's a whitespace change somewhere here which is causing it to show the whole file as new lines. Can you delete this file and edit an original one again?

Reply:
Problem is that I'm not able to see where the code coverage is dropping on codecov.io